resources/session03/wsgi/calculator.py

```diff
@@ -0,0 +1,66 @@
+import re
+
+
+def resolve_path(path):
+    urls = [(r'^$', welcome),
+            (r'^add/(\d+)/(\d+)$', add),
+            (r'^subtract/(\d+)/(\d+)$', subtract),
+            (r'^divide/(\d+)/(\d+)$', divide),
+            (r'^multiply/(\d+)/(\d+)$', multiply)]
+    
+    matchpath = path.lstrip('/')
+    for regexp, func in urls:
+        match = re.match(regexp, matchpath)
+        if match is None:
+            continue
+        args = match.groups([])
+        return func, args
+    # we get here if no url matches
+    raise NameError
+
+def welcome():
+    return "<h1>welcome</h1>"
+
+def multiply(v1, v2):
+    r = int(v1) * int(v2)
+    return "<h1>result from multiplication: %d</h1>" %r
+
+def add(v1, v2):
+    r = int(v1) + int(v2)
+    return "<h1>result from addition: %d</h1>" % r  
+
+def subtract(v1, v2):
+    r = int(v1) - int(v2)
+    return "<h1>result from subtraction: %d</h1>" % r
+
+def divide(v1, v2):
+    r = int(v1) / int(v2)
+    return "<h1>result from division: %d</h1>" % r
```

Comment:
Do the four functions you've written here feel a bit repetitive to you?  They do to me.  Notice how each function is really identical with the exception of the operator being used, and the name of the operator in the message returned?  Can you think of a way to compact that a bit?  Take a look at the [operator module](https://docs.python.org/3.5/library/operator.html) in the Python standard library.  Does knowing that exist help you to see a way to a less repetitive solution?